pkg/proxy/iptables/proxier.go

```diff
@@ -1306,17 +1306,6 @@ func (proxier *Proxier) syncProxyRules() {
 					utilproxy.WriteLine(proxier.natRules, append(args, "-j", string(KubeMarkMasqChain))...)
 					// Jump to the service chain.
 					utilproxy.WriteLine(proxier.natRules, append(args, "-j", string(svcChain))...)
-				} else {
-					// TODO: Make all nodePorts jump to the firewall chain.
-					// Currently we only create it for loadbalancers (#33586).
-
-					// Fix localhost martian source error
-					loopback := "127.0.0.0/8"
-					if isIPv6 {
-						loopback = "::1/128"
-					}
-					utilproxy.WriteLine(proxier.natRules, append(args, "-s", loopback, "-j", string(KubeMarkMasqChain))...)
-					utilproxy.WriteLine(proxier.natRules, append(args, "-j", string(svcXlbChain))...)
```

Comment:
@thockin this was driving me crazy because it turns out that NodePorts on nodes without local endpoints, still forward traffic if the connection comes from one pod.
I don't know if that is on purpose, can you chime in please?

Reply:
this was ok, I misread it